File: clock/abin_abraham/use this folder for TESTing my code/clock.java
import java.applet.*;
import java.awt.*;
import java.awt.event.*;
import java.util.*;
import java.util.Date;
import java.text.DateFormat;
import java.lang.*;

/*<applet code="clock.class" height=100 width=400>
</applet>*/


public class clock extends Applet implements Runnable
{


int hrs,mins,flag=0;
String h="",m="";
String hrstring[]={"one","two","three","four","five","six","seven","eight","nine","ten","eleven","twelve"};
String minstring[]={"five","ten","quarter","twenty","twenty-five","half-past"};

String near[]={"after ","about ","around "};
String correct[]={"exactly ","precisely ","now ",""};
String far[]={"almost ","nearly "};

String disp="";


String time="";
Date date=new Date();
DateFormat format= DateFormat.getTimeInstance();


Random rand=new Random();
int r=rand.nextInt(3);

Thread timerthread=null;






public void getDisp()
{
disp="its ";

hrs=date.getHours();
mins=date.getMinutes();


if(hrs>12)
hrs-=12;
else if(hrs==0)
hrs=12;



if((mins==0)||(mins==15)||(mins==30)||(mins==45))
{
disp+=correct[rand.nextInt(4)];
}



if(mins==15)
{
disp+="quarter past ";
disp+=hrstring[hrs-1];
}
else if(mins==30)
{
disp+="half-past ";
disp+=hrstring[hrs-1];
}
else if(mins==45)
{
disp+="quarter to ";
disp+=hrstring[hrs];
}
else if(mins==0)
{
disp+=hrstring[hrs-1];
disp+=" o'clock";
}


else if((mins%5==0)&&(mins<30))
{
disp+=correct[rand.nextInt(4)];
disp+=minstring[mins/5-1];
disp+=" past ";
disp+=hrstring[hrs-1];
}
else if((mins%5==0)&&(mins>30))
{
disp+=correct[rand.nextInt(4)];
disp+=minstring[12-mins/5-1];
disp+=" to ";
disp+=hrstring[hrs];
}



//upto 30 minutes

else if(mins<3)
{
disp+="just past ";
disp+=hrstring[hrs-1];
}



else if(((mins%10<3)||((mins%10>5)&&(mins%10<8)))&&(mins<30))
{
disp+=near[rand.nextInt(3)];
disp+=minstring[mins/5-1];
disp+=" past ";
disp+=hrstring[hrs-1];
}
else if((((mins%10>=3)&&(mins%10<5))||(mins%10>=8))&&(mins<30))
{
disp+=far[rand.nextInt(2)];
disp+=minstring[mins/5];
disp+=" past ";
disp+=hrstring[hrs-1];
}




//above 30 minutes

else if(mins>=58)
{
disp+="almost ";
disp+=hrstring[hrs];
}
else if((mins>30)&&(mins<33))
{
disp+="just after ";
disp+=minstring[12-mins/5-1];
disp+=" past ";
disp+=hrstring[hrs-1];
}



else if(((mins%10<3)||((mins%10>5)&&(mins%10<8)))&&(mins>30))
{
disp+=near[rand.nextInt(3)];
disp+=minstring[12-(mins/5)-1];
disp+=" to ";
disp+=hrstring[hrs];
}
else if((((mins%10>=3)&&(mins%10<5))||(mins%10>=8))&&(mins>30))
{
disp+=far[rand.nextInt(2)];
disp+=minstring[12-(mins/5)-2];
disp+=" to ";
disp+=hrstring[hrs];
}




}











public void start()
{
timerthread=new Thread(this,"clock");
timerthread.start();
}


public void stop()
{
if(timerthread==null)
return;
timerthread=null;
}


public void run()
{
while(timerthread!=null)
{


repaint();


try
{
timerthread.sleep(501);
}
catch(InterruptedException e)
{
System.out.println("Exception: "+e);
}
}
}


public void paint(Graphics g)
{

date.setTime(System.currentTimeMillis());
String time=format.format(date);
if((date.getSeconds()==0)||flag==0)
{
getDisp();
flag=1;
}
g.drawString(time,50,20);
g.drawString(disp,50,40);


}


}

File: clock/paul_philip/clock.py
# -*- coding: utf-8 -*-

import random
import sys
import time
from datetime import datetime

HOURS = ['twelve','one','two','three','four','five','six',
         'seven','eight','nine','ten','eleven']
MINUTES = {
  '0':  '',  
  '5':  'five past',
  '10': 'ten past',
  '15': 'quarter past',
  '20': 'twenty past',
  '25': 'twenty-five past',
  '30': 'half past',
  '35': 'twenty-five to',
  '40': 'twenty to',
  '45': 'quarter to',
  '50': 'ten to',
  '55': 'five to',
  '60': ''
}
PREPOSITIONS = {
  '-1': ['almost', 'nearly'],
  '0': ['exactly', 'precisely', 'now', ''],
  '1': ['after', 'about', 'around']
}
roundAbout = 'it’s ’round about midnight.'

SPECIAL_CASES = {
  '23:58': roundAbout,
  '23:59': roundAbout,
  '00:00': 'it’s midnight.',
  '00:01': roundAbout,
  '00:02': roundAbout,
  '12:00': 'it’s noon.'
}

def print_time(hrs,mins,prep):
    if mins==0:
        print "It’s %s o’clock." %(HOURS[hrs])
    else:
        print "Its %s %s %s" %(PREPOSITIONS[str(prep)][random.randrange(0,len(PREPOSITIONS[str(prep)]))],MINUTES[str(mins)],HOURS[hrs])



def system_time():
    local=str(datetime.time(datetime.now()))
    return local[0:5]

def time_conv(tim_word):
    t=time.strptime(tim_word,"%H:%M")
    return t

def main():
    f=0
    if len(sys.argv)==1:
        clock=system_time()
        print clock
        a=time_conv(clock)
    else:
        clock=sys.argv[1]
        print clock
        a=time_conv(clock)
        

    hrs=a.tm_hour
    mins=a.tm_min

    if SPECIAL_CASES.has_key(clock):
        print SPECIAL_CASES[clock]
        sys.exit(0)
    if hrs > 12:
        hrs=hrs-12
    if mins==30 or mins ==0:
        prep=0
    elif mins > 30:
        prep=-1
        hrs=hrs+1
    else:
        prep=1

    k=mins%5
    if k==0:
        mins=mins
    elif k > 2.5:
        mins=int((mins/5)+1)*5
    elif k<2.5:
        mins=int(mins/5)*5
    if mins==60:
        mins=55
    print_time(hrs,mins,prep)

if __name__ == "__main__":
    main()





